package bnet.protocol.friends;
import "bnet/attribute.proto";
import "bnet/entity.proto";
import "bnet/invitation_types.proto";


message Friend {
	required EntityId id = 1;
	repeated attribute.Attribute attribute = 2;
	repeated uint32 role = 3 [packed = true];
	optional uint64 privileges = 4 [default = 0];
	optional uint64 attributes_epoch = 5;
	optional string full_name = 6;
	optional string battle_tag = 7;
}

message FriendInvitation {
	optional bool first_received = 1 [default = false];
	repeated uint32 role = 2 [packed = true];
	extend .bnet.protocol.invitation.Invitation {
		optional FriendInvitation friend_invite = 103;
	}
}

message FriendInvitationParams {
	optional string target_email = 1;
	optional string target_battle_tag = 2;
	optional string inviter_battle_tag = 3;
	optional string inviter_full_name = 4;
	optional string invitee_display_name = 5;
	repeated uint32 role = 6 [packed = true];
	extend .bnet.protocol.invitation.InvitationParams {
		optional FriendInvitationParams friend_params = 103;
	}
}
